SWENT-GRP5/SOLVIT
# Generalize service requests

### Description:  
This pull request generalizes and enhances the `ServiceRequest` model while refactoring related functionality across the app. It updates the `JobDashboardScreen` to use the `ServiceRequest` instead of  the previously used `Job` class. `ServiceRequest`Redundant files and references have been cleaned up, and was also modified.  

- **What issue does it solve?** This PR unifies service request handling by eliminating the `Job` class, improving maintainability and clarity.
- **What functionality does it add?** The `ServiceRequest` model now fully replaces `Job`, providing a streamlined approach to handling service requests, dynamic updates for request statuses, and improved consistency across the codebase.  

### Related Issues:  
- #331 

### Changes Made:  
- Generalized `ServiceRequest` to support additional information and extended its use across the app.  
- Updated `ServiceRequestRepositoryFirebase` to accommodate the modified model.  
- Added methods to the MVVM structure to replace the `Job` class with `ServiceRequest`.  
- Refactored `JobDashboardScreen` to use `ServiceRequest`.  
- Enabled dynamic updates to the request lists for each status.  
- Updated tests to align with the new structure and removed redundant `Job` ViewModel files.  
- Standardized file names for improved readability and consistency.  

### How to Test:  
1. Run the updated tests to validate functionality and ensure no regressions.  
2. Navigate the app to ensure that everything works as intended.
3. Review the file names to confirm consistency with the updated naming.  

---

### Checklist:
